### ISSUE TITLE
'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.

### ISSUE THREAD

**Issue by mykhailo-melnyk** [facebook/react-native#26987]
I tried to add new schemas to ios project. I added Staging-Debug copied from default Debug schema. When application is loading I catch the next error:

```
2019-10-24 15:43:05.427 [error][tid:com.facebook.react.JavaScript] Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.
2019-10-24 15:43:05.430 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.
2019-10-24 15:43:05.433396+0300 HelloGuest[3076:1701296] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.', reason: 'Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the na..., stack:
getEnforcing@4947:28
<unknown>@41941:50
loadModuleImplementation@271:14
<unknown>@41900:40
loadModuleImplementation@271:14
<unknown>@36481:18
loadModuleImplementation@271:14
<unknown>@29159:16
loadModuleImplementation@271:14
guardedLoadModule@163:47
global code@375133:4

```

React Native version:

  System:
    OS: macOS 10.15
    CPU: (4) x64 Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
    Memory: 110.47 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.0, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    Android SDK:
      API Levels: 23, 24, 26, 27, 28
      Build Tools: 23.0.1, 27.0.3, 28.0.2, 28.0.3
      System Images: android-24 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 11.0/11A420a - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.2 => 0.61.2
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-create-library: 3.1.2
    react-native-git-upgrade: 0.2.7

Why can it happen?

**Comment by anhkieet** [comment]
I have the same issue with you :(

**Comment by matthiasthomas** [comment]
Same here!

**Comment by maullerz** [comment]
Same here!

**Comment by maxperry** [comment]
Same here!

**Comment by maullerz** [comment]
In my case, i got this error only when trying to run debug build on device.
If i try to run on simulator - i am getting "No Bundle URL Present."
It is not connect to development server for js bundle.
After debugging finally realized that RCT_DEV not setted for debug builds in my new schema.

After some workaround like adding `RCT_DEV=1` in Preprocessor flags,
started to get only error "TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found."
in simulator also.

**Comment by maullerz** [comment]
If it helps,
for 0.61.2 there is no DevSettings.js in
./node_modules/react-native/Libraries/Utilities/

**Comment by maullerz** [comment]
Workaround for me
1. renamed back my "DEMO Debug" config to "Debug"
2. changed my schemes accordingly
3. also fixed "React(missing)" in Build part of schemes.

![Снимок экрана 2019-10-29 в 12 31 00](https://user-images.githubusercontent.com/5295025/67755417-27df1a00-fa49-11e9-9c7d-708aa1988b6e.png)

Now debug build working as expected, but i cant use different configs for different schemas, only 1 "Debug".

**Comment by mykhailo-melnyk** [comment]
@maullerz yes, you are right but I want to use a different configuration for different schema
=(

**Comment by Jonovono** [comment]
Getting this too, any ideas?

```
Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.
```

**Comment by matthiasthomas** [comment]
>Getting this too, any ideas? ...

@Jonovono so in my case I was actually only getting this on the Staging scheme, Local and Prod worked fine.
The issue was that debug was on for Staging, even though it’s supposed to behave like a release build. Now that I build Staging release it works fine.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by corneelius** [comment]
Make sure your bundle command includes --dev false if you're building for release

For example:
react-native bundle --entry-file index.js --dev false --reset-cache --bundle-output main.jsbundle

**Comment by mykhailo-melnyk** [comment]
@corneelius where can I find these settings if I run project using Xcode?

**Comment by corneelius** [comment]
In Build Phases, there is a phase called Bundle React Native Code and Images. That runs a script that creates the bundle for release builds. It fails for me most of the time. I commented it out and created the bundle manually by running this: 

react-native bundle --entry-file index.js --dev false --reset-cache --bundle-output main.jsbundle

After that, drag the bundle into the Xcode project and try running again in release mode.

**Comment by mykhailo-melnyk** [comment]
any updates on this?

**Comment by takameyer** [comment]
I was able to solve this in my Podfile
```ruby
target 'MyApp' do
  project 'MyApp',
  'Staging-Debug' => :debug,
  'Staging-Release' => :release,
  ...
```
This ensures that your build configuration is marked as "debug" or "release" in the underlying Pod xcode projects (including react-native).

**Comment by arbielsk** [comment]
@takameyer Thank you for your solution, this works perfectly for my version of the issue! 

**For anyone with a similar issue to mine, this might be helpful:**
I've been on this issue way too long updating from 0.59 -> 0.61.4 and switching from manually linked packages to using cocoapods.

In my case the issue already presented it before this with 'No bundle url present' because preprocessing-variable RCT_DEV was automatically set to 0. Only after finding a workaround for that issue (setting the jsBundleLocation in AppDelegate.m to a hardcoded value in case it was nil) I got the issue seen above.

We are using react-native-schemes-manager which previously solved this issue automagically.

**Comment by mykhailo-melnyk** [comment]
@takameyer 
Could you please show full pod file?
What does this sentence mean?
"This ensures that your build configuration is marked as "debug" or "release" in the underlying Pod xcode projects (including react-native)."

**Comment by takameyer** [comment]
@mykhailo-melnyk 
1. No Problem:
```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

target 'MyApp' do
  project 'MyApp', 
  'Staging-Debug' => :debug,
  'Staging-Release' => :release,

  # Pods for HBAPP
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'
  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  use_native_modules!

end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'react-native-config'
      phase = target.project.new(Xcodeproj::Project::Object::PBXShellScriptBuildPhase)
       phase.shell_script = "cd ../../"\
        " && RNC_ROOT=./node_modules/react-native-config/"\
        " && export SYMROOT=$RNC_ROOT/ios/ReactNativeConfig"\
        " && ruby $RNC_ROOT/ios/ReactNativeConfig/BuildDotenvConfig.ruby"
 
       target.build_phases << phase
       target.build_phases.move(phase,0)
    end
  end
end
```

2. I was previously using react-native-scheme-manager to do this in older version of RN.  What this did was go through every xcodeproj file that existed in node_modules and wrote all my projects build configurations in them (at least that is my rough understanding). 
What the Podfile does is quite similar, except it does no rewriting.  It basically says take the build configuration "Staging-Debug" and handle it as if it were the build configuration "debug" in all the Pod dependencies.  That is also only my rough understanding of how it works.

3. Yes. I would recommend a complete reinstall of everything:
```bash
> rm -rf node_modules
> yarn
> cd ios
> pod install
```

Another caveat, if you, like me, are coming from using 'react-native-scheme-manager", uninstall this completely using the instructions from https://github.com/thekevinbrown/react-native-schemes-manager#uninstalling
In my case this was no longer necessary to have and only caused more problems migrating from RN 59 -> 61

**Comment by mykhailo-melnyk** [comment]
@takameyer 
thanks a lot!
it really helped in my case ( I also removed using react-native-scheme-manager)

**Comment by luskin** [comment]
Unfortunately we are still receiving this error. We are building a fresh app on `0.61.4`.

Podfile:
```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

target 'Bolt' do
  # Custom Configs
  project 'MyApp',
  'Debug (Staging)' => :debug,
  'Debug (Production)' => :debug,
  'Release (Staging)' => :release,
  'Release (Production)' => :release
  
  # Pods for MyApp
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  
  # Firebase
  pod 'RNFBApp', :path => '../node_modules/@react-native-firebase/app'
  pod 'RNFBDatabase', :path => '../node_modules/@react-native-firebase/database'
  pod 'RNFBAuth', :path => '../node_modules/@react-native-firebase/auth'
  
  pod 'ReactNativeNavigation', :podspec => '../node_modules/react-native-navigation/ReactNativeNavigation.podspec'
  
  pod 'react-native-intercom', :path => '../node_modules/react-native-intercom'
  
  ## PERMISSIONS
  permissions_path = '../node_modules/react-native-permissions/ios'
  pod 'Permission-Camera', :path => "#{permissions_path}/Camera.podspec"
  pod 'Permission-LocationAlways', :path => "#{permissions_path}/LocationAlways.podspec"
  pod 'Permission-Notifications', :path => "#{permissions_path}/Notifications.podspec"
  pod 'Permission-PhotoLibrary', :path => "#{permissions_path}/PhotoLibrary.podspec"

  use_native_modules!
end
```

We have done several full removals and reinstalls of the pods including the lockfile. Not sure where to go from here, any help would be _greatly_ appreciated.

**Comment by takameyer** [comment]
@luskin How were your build configurations generated?  Have you tried making a new one by duplicating 'Debug' and 'Release' from the Project -> Info tab?
![image](https://user-images.githubusercontent.com/164606/69042144-02628200-09f1-11ea-8142-5274ac2d9215.png)
I would attempt this again and try and build only with that configuration.  If that works then generate the rest in the same manner.

**Comment by luskin** [comment]
@takameyer Yes, this is how I created the additional configurations. I duplicated Debug and Release to create the new Staging and Production debug and release configurations.

**Comment by RSNara** [comment]
This error occurs because the native implementation of "ExceptionsManager" wasn't found. As I'm converting these NativeModules to TurboModules, anything that lies in these paths will be moved to CoreModules:

```
[
    "React/Base/**/*.m",
    "React/Base/**/*.mm",
    "React/DevSupport/**/*.m",
    "React/DevSupport/**/*.mm",
    "React/Inspector/**/*.m",
    "React/Inspector/**/*.mm",
    "React/Modules/**/*.m",
    "React/Modules/**/*.mm",
    "React/Profiler/**/*.m",
    "React/Profiler/**/*.mm",
    "React/Profiler/**/*.S",
    "React/UIUtils/*.m",
    "React/Views/**/*.m",
    "React/Views/**/*.mm",
    "Libraries/ActionSheetIOS/*.m",
    "Libraries/WebSocket/*.m",
]
```

To anyone who has this problem, does depending on the `React-CoreModules` spec fix the problems?

**Comment by luskin** [comment]
@RSNara to confirm, you are recommending updating the Podfile to include:

`pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'`

If so I will re-update the app to RN `0.61.4`, include this pod and report back

**Comment by ahartzog** [comment]
@luskin @RSNara I've had the React-CoreModule in my pod since September 18th, both before and after experiencing this issue. 
![image](https://user-images.githubusercontent.com/9828263/69289075-63dd4900-0bc9-11ea-8ec9-aeede55f6225.png)

**Comment by treyp** [comment]
I had this problem, but turns out I wasn't using the correct build configuration name.

If you're working on a brownfield app like me, here's where you can see the exact build configuration names in the scheme editor:

<img width="1034" alt="Screen Shot 2019-11-20 at 12 01 06 PM" src="https://user-images.githubusercontent.com/49319/69289110-207edc80-0bc1-11ea-87b2-c41758c7589a.png">

Make sure they match up exactly in your podfile:

![Screen Shot 2019-11-20 at 12 03 01 PM](https://user-images.githubusercontent.com/49319/69289122-28d71780-0bc1-11ea-8aaa-06d5a58763a3.png)

**Comment by luskin** [comment]
@treyp - We reverted any custom schemes and are back to the classic Debug/Release schemes, so this doesn't apply to us.

@RSNara - We upgraded back up to `0.61.4` from `0.60.5` to re-attempt your proposed solution but it does not work. We are still receiving the error.

**Comment by ahartzog** [comment]
Removing the --development flag from my codepush release build process resolved this error for me.

**Comment by luskin** [comment]
@ahartzog You seemed to have solved the issue! We were using the `--dev` flag to deploy to our staging environment but removing that and setting the destination like you advised solved the issue. THANK YOU!

**Comment by ahartzog** [comment]
Awesome, glad to hear that. I'd write a PR to update the docs that this is an issue with 0.61+ but...it wouldn't get merged so...

**Comment by kmvkrish** [comment]
You can follow this link for a clear explanation: [Multiple schemes in RN apps](https://medium.com/@ywongcode/building-multiple-versions-of-a-react-native-app-4361252ddde5)

**Comment by nihp** [comment]
I am getting the same error. How can I resolve this? I tried the following 

react-native bundle --entry-file ./index.js --platform ios --bundle-output ios/main.jsbundle --assets-dest ios

But it not worked for me. Again I am getting 

**'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.**

**Comment by nihp** [comment]
ahartzog Where i need to remove the development flag

**Comment by rbarbish** [comment]
@takameyer - Show your package.json

**Comment by nihp** [comment]
The above is fixed for me after changing the node -v as before.

I have used node -v 9.4.0 and again changed to 12.0. So I got above error.

To resolve this I have moved to the old one 9.4.0 and it get solved

**Comment by alpeshgaglani** [comment]
Any updates on this issue? It's preventing us from using RN in our existing app - Bundling and accessing it as an asset works, but it's hurting productivity big time because we can load from local metro bundler service, or do any debugging.

**Comment by ivanushka1703** [comment]
I was able to solve this problem, move custom configurations in Podfile above target section.
![image](https://user-images.githubusercontent.com/23135685/83410772-b2448d80-a41f-11ea-819a-902105968ba5.png)

**Comment by cbedoy** [comment]
I'm not using custom configurations, we only have Debug|Release schemas.

I've been trying with 0.62.2, and 0.63.2, but still having instant crash when run on release mode or when testers got new build from testflight.


```
2020-08-04 10:14:38.514908-0500 RNCliDagM8[67965:1042593] [] nw_socket_handle_socket_event [C3.1:1] Socket SO_ERROR [61: Connection refused]
2020-08-04 10:14:38.515 [error][tid:com.facebook.react.JavaScript] Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.
2020-08-04 10:14:38.516991-0500 RNCliDagM8[67965:1042593] [] nw_socket_handle_socket_event [C3.2:1] Socket SO_ERROR [61: Connection refused]
2020-08-04 10:14:38.518256-0500 RNCliDagM8[67965:1042584] [] nw_connection_get_connected_socket [C3] Client called nw_connection_get_connected_socket on unconnected nw_connection
2020-08-04 10:14:38.518556-0500 RNCliDagM8[67965:1042584] TCP Conn 0x600003a84e40 Failed : error 0:61 [61]
2020-08-04 10:14:38.519 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.
2020-08-04 10:14:38.552387-0500 RNCliDagM8[67965:1042593] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.', reason: 'Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the na..., stack:```

**Comment by sameid** [comment]
I've been experiencing the same issue since I updated to Xcode 12.

Here is my react-native info output:
```
System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 4.21 GB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.2 - ~/.nvm/versions/node/v12.16.2/bin/node
    Yarn: Not Found
    npm: 6.14.4 - ~/.nvm/versions/node/v12.16.2/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.0, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
    Android SDK:
      API Levels: 26, 27, 28, 29
      Build Tools: 27.0.3, 28.0.0, 28.0.2, 28.0.3, 29.0.3
      Android NDK: Not Found
  IDEs:
    Android Studio: 3.6 AI-192.7142.36.36.6392135
    Xcode: 12.0/12A7209 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_251 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.2 => 0.63.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Here is what happens when I try to run the scheme in Release mode.

```
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the native binary.', reason: 'Unhandled JS Exception: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'DevSettings' could not be found. Verify that a module by this name is registered in the na..., stack:
invariant@4555:25
getEnforcing@4826:27
<unknown>@35255:49
loadModuleImplementation@271:13
<unknown>@35165:61
loadModuleImplementation@271:13
<unknown>@34910:31
loadModuleImplementation@271:13
<unknown>@35943:35
loadModuleImplementation@271:13
<unknown>@27866:15
loadModuleImplementation@271:13
guardedLoadModule@163:46
global code@180966:3
'
terminating with uncaught exception of type NSException
(lldb) 
```

Important note is that I am able to run my build in Debug mode.

**Comment by dehypnosis** [comment]
@sameid didnt work for me..

**Comment by dehypnosis** [comment]
A workaround.

1. Comment iOS invariant assertion in `NativeEventEmitter.js`
```diff
diff --git a/node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter.js b/node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter.js
index 59e03aa..1a6bdbf 100644
--- a/node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter.js
+++ b/node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter.js
@@ -34,7 +34,9 @@ class NativeEventEmitter extends EventEmitter {
   constructor(nativeModule: ?NativeModule) {
     super(RCTDeviceEventEmitter.sharedSubscriber);
     if (Platform.OS === 'ios') {
-      invariant(nativeModule, 'Native module cannot be null.');
+      // COMMENTED AS PATCH: .... hell noway to fix DevSettings bug
+      // ref: https://github.com/facebook/react-native/issues/27758
+      // invariant(nativeModule, 'Native module cannot be null.');
       this._nativeModule = nativeModule;
     }
   }
```

2. in iOS `Pods` project `Build Settings`, Set `Processor Macros` for `Release` as like
```
...
RCT_DEBUG=0
RCT_DEV=0
RCT_DEV_MENU=1
ENABLE_PACKAGER_CONNECTION=0
```

Hmm... It works for now.
My current env:
```
$ react-native info
info Fetching system and libraries information...
System:
    OS: macOS 10.15.6
    CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
    Memory: 982.60 MB / 40.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.2 - /usr/local/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0. - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.7, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 23, 26, 27, 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.2
      System Images: android-26 | Intel x86 Atom_64, android-26 | Google APIs Intel x86 Atom, android-27 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.2 Canary 11 4.2 Canary 11
    Xcode: 11.7/11E801a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.5 - /Users/dw.kim/.sdkman/candidates/java/current/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: ^0.63.3 => 0.63.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found (It is )
```

**Comment by jamesjara** [comment]
Hello, this error took me a lot  of time as well :( 
so let me share the my solution, the problem is with the `node_modules/react-native/scripts/react-native-xcode.sh` for some reason is passing `--dev true` even in the release configurations. So i did a patched copy of the `react-native-xcode.sh` and  copy it in the postintall script" 

```
"$NODE_BINARY" $NODE_ARGS "$CLI_PATH" $BUNDLE_COMMAND \
  $CONFIG_ARG \
  --entry-file "$ENTRY_FILE" \
  --platform ios \
  --dev false \. // pass as false dev
  --reset-cache \
  --bundle-output "$BUNDLE_FILE" \
  --assets-dest "$DEST" \
  $EXTRA_PACKAGER_ARGS
```

**Comment by evanjmg** [comment]
This issue can be closed since it's podfile configuration - might want to add this to setup instructions

**Comment by elsurudo** [comment]
This is still an issue, and I don't think it's just podfile configuration.

In my case, I can see in my build output that the command being run is correct (e.g. `--dev false`). Despite this, I am _still_ getting the same error. Does anyone have any idea what else to try here? I tried pretty much all the applicable solutions in this thread to no avail...

**Comment by elsurudo** [comment]
Well, I feel pretty dumb, but in my case it ended up being that _I was using DevSettings in my index.js file_, so I needed to make sure that code only ran in dev, like so:

```
if (Config.ENV == 'development') {
  ...
}
```

**Comment by Celleb** [comment]
This is just a workaround that might provide some relief. This worked on RN v0.62 but not RN v0.64

Find this file `node_modules/react-native/Libraries/Utilities/DevSettings.js`

comment out this line or delete this line `import NativeDevSettings from '../NativeModules/specs/NativeDevSettings';`

comment out or delete this class:

```js
class DevSettings extends NativeEventEmitter implements IDevSettings {
  _menuItems: Map<string, () => mixed>;

  constructor() {
    super(NativeDevSettings);

    this._menuItems = new Map();
  }

  addMenuItem(title: string, handler: () => mixed) {
    // Make sure items are not added multiple times. This can
    // happen when hot reloading the module that registers the
    // menu items. The title is used as the id which means we
    // don't support multiple items with the same name.
    const oldHandler = this._menuItems.get(title);
    if (oldHandler != null) {
      this.removeListener('didPressMenuItem', oldHandler);
    } else {
      NativeDevSettings.addMenuItem(title);
    }

    this._menuItems.set(title, handler);
    this.addListener('didPressMenuItem', event => {
      if (event.title === title) {
        handler();
      }
    });
  }

  reload(reason?: string) {
    if (typeof NativeDevSettings.reloadWithReason === 'function') {
      NativeDevSettings.reloadWithReason(reason ?? 'Uncategorized from JS');
    } else {
      NativeDevSettings.reload();
    }
  }

  onFastRefresh() {
    if (typeof NativeDevSettings.onFastRefresh === 'function') {
      NativeDevSettings.onFastRefresh();
    }
  }

  // TODO: Add other dev setting methods exposed by the native module.
}
```
modify this line 
```js
module.exports = ((__DEV__
  ? new DevSettings()
  : new NoopDevSettings()): IDevSettings);
```

To
```js
module.exports = (new NoopDevSettings(): IDevSettings);
```

**Comment by LordMendes** [comment]
In my specific situation, I was using a deprecated version of expo/ex-navigation , in RN 0.64.2 we had one problem, because RN doesn't understood the @2x.android or similars casts to choose the right image for an Icon. The problem is Described here : [#31348](https://github.com/facebook/react-native/issues/31348).

I sold my conflicts locally inside node_modules and make Android work fine (then I forked and create a correction under my control), but when I came back to work in my IoS problems, I forgot to use this correction.

So, my problem was occurring in one lib i had in my node_modules, and as I extensively searched, this problems use to happen in this kind of situation.

So if @takameyer solution doesn't work as it doesn't to me, I suggest taker a look in your node_modules dependencies...

probably trying to build your bundle thought react-native-cli will give you an better log then building this in xcode, since in my case xcode build worked but give-me a run-time error(Release).

I hope it help some one in the future.
